dev/tasks/jars/build-cpp-osx.sh

```diff
@@ -17,9 +17,11 @@
 # specific language governing permissions and limitations
 # under the License.
 
-set -e
+set -ex
```

Comment:
We maintain the OSX wheel scripts next to the manylinux ones in `ci/scripts`. Could you please move this file there as well for consistency?

Something like:

```
ci/scripts/java_<whatever>_manylinux_build.sh
ci/scripts/java_<whatever>_macos_build.sh
```